### ISSUE TITLE
Make new CrossLinkingMassSpectrometryRestraint compatible with PMI2

### ISSUE THREAD

**Issue by cgreenberg** [salilab/pmi#159]
Not too bad, just need to make representation optional (pass root hierarchy instead) and use IMP Selection in that case. The interface will slightly change from:

``` python
__init__(self, representation,
                 CrossLinkDataBase,
                 length,
                 resolution=None,
                 slope=0.02,
                 label="None",
                 filelabel="None",
                 attributes_for_label=None):
```

to

``` python
__init__(self, representation=None,
                 root_hier = None
                 CrossLinkDataBase = None,
                 length = 10.0,
                 resolution=None,
                 slope=0.02,
                 label="None",
                 filelabel="None",
                 attributes_for_label=None):
```

so you'll have to explicitly pass any arguments if you were previously just passing `(representation, XLDB)`

**Comment by cgreenberg** [comment]
This actually won't work with ambiguity. Currently the restraint just loops over every XL with the same unique_id and adds a contribution from that. It only expects ONE particle pair per XL. The idea is that if you have an ambiguous system, you need to first clone the protein multiple times in the database object (ProtA -> ProtA.1, ProtA.2). You need this because in PMI1, the copies are only related via names.

However in PMI2 it's easier to find copies (in fact we just use the Copy decorator). So for example you can get all ambiguous copies with a simple selection:

``` python
sel = IMP.atom.Selection(hier,residue_index=5,molecule="ProtA",state_index=0)
```

If there are 10 copies of this molecule, they're all returned. Cool! But this isn't compatible with the database "cloning" concept.

Basically the only option here is to internally create new XLs within the database (same unique id) for each ambiguous pair. Hopefully this will work fine to have multiple XL entries with the exact same molecule name and residue index.

We really need to stop developing PMI1, this is getting ridiculous....

**Comment by cgreenberg** [comment]
Here is my plan: 
- The CrossLinkingMassSpectrometryRestraint needs to first search the list of molecule names for available copies in the root hierarchy. (I guess with a quick selection).
- For each Copy, make a named clone in the database ("ProtA.0, ProtA.1, ...")
- Then when looping through unique ids, if it's PMI2, it needs to extract the copy number from the name in order to do selection properly. 

A better solution would be to use a "copy_number" key in the components XL Database but I have no idea how to do that. @Pellarin any ideas ?
